hardware_interface/src/fake_components/generic_system.cpp

```diff
@@ -257,28 +314,42 @@ std::vector<hardware_interface::CommandInterface> GenericSystem::export_command_
   // Fake sensor command interfaces
   if (fake_sensor_command_interfaces_)
   {
-    for (auto i = 0u; i < info_.sensors.size(); i++)
-    {
-      const auto & sensor = info_.sensors[i];
-      for (const auto & interface : sensor.state_interfaces)
-      {
-        if (!get_interface(
-              sensor.name, sensor_interfaces_, interface.name, i, sensor_fake_commands_,
-              command_interfaces))
-        {
-          throw std::runtime_error(
-            "Interface is not found in the standard nor other list. "
-            "This should never happen!");
-        }
-      }
-    }
+    populate_command_interfaces(
+      "standard nor other", info_.sensors, sensor_interfaces_, sensor_fake_commands_,
+      command_interfaces);
+  }
+
+  // Fake gpio command interfaces
+  if (fake_gpio_command_interfaces_)
+  {
+    populate_command_interfaces(
+      "gpio", info_.gpios, gpio_interfaces_, gpio_fake_commands_, command_interfaces);
+  }
+  // GPIO command interfaces (real commands)
+  else
+  {
+    populate_command_interfaces(
+      "gpio", info_.gpios, gpio_interfaces_, gpio_commands_, command_interfaces);
   }
 
   return command_interfaces;
 }
 
 return_type GenericSystem::read()
 {
+  auto mirror_command_to_state = [](size_t start_index, auto & states_, auto commands_) {
```

Comment:
not sure we need the "start_index_"

Reply:
This I added due to the below line. Else we don't need this start_index. https://github.com/ros-controls/ros2_control/blob/41f3cce1a9c3c6ec23ea51a19c4a58d65e0ad70a/hardware_interface/src/fake_components/generic_system.cpp#L364

Reply:
This is OK, but then you should use it.

Reply:
This is being used in the next line,

for (size_t i = start_index; i < states_.size(); ++i)

Reply:
Please change the function to the following prototype: (use default value for start index)

```
auto mirror_command_to_state = [](auto & states_, const auto & commands_, size_t start_index = 0)
```